internal/parse/parser.go

```diff
@@ -278,6 +304,138 @@ func (p *Parser) parseGoFullName() (FullName, bool, error) {
 	return FullName{}, false, nil
 }
 
+// parseList takes a parsing function that returns a FullName and parses a
+// bracketed, comma seperated, list. On success it returns an array of FullName
+// objects. Otherwise it returns an empty array, false and an error.
+func (p *Parser) parseList(parseFn func(p *Parser) (FullName, bool, error)) ([]FullName, bool, error) {
+	cp := p.save()
+	if p.skipByte('(') {
+		parenPos := p.pos
+		p.skipSpaces()
+
+		nextItem := true
+		var objs []FullName
+		for nextItem {
+			if obj, ok, err := parseFn(p); ok {
+				objs = append(objs, obj)
+				p.skipSpaces()
+			} else if err != nil {
+				return nil, false, err
+			} else {
+				return nil, false, fmt.Errorf("invalid identifier near char %d", p.pos)
+			}
+			p.skipSpaces()
+			if p.skipByte(')') {
+				return objs, true, nil
+			}
+			nextItem = p.skipByte(',')
```

Comment:
Should you skip spaces before the `','`? Having a space there would be bad formatting, but might not be considered a syntax error

Reply:
We do, on line 327

Reply:
327 checks for a space before the bracket (` ),`), this code doesn't allow for whitespace between the bracket and comma (`) ,`) :P

Reply:
I'm not sure I understand, once we parse a closing bracket we return. When the bracket is closed we finish parsing, we want to ignore everything after the bracket in this function and leave it up to the caller how they want to handle it.

Reply:
Ah right my bad I misread, this is fine